src/snowcli/cli/nativeapp/manager.py

```diff
@@ -35,3 +66,86 @@ def deploy_root(self) -> Path:
+
+    def sync_deploy_root_with_stage(self, role: str, app_pkg: str):
+        schema_and_stage: str = self.definition["source_stage"]
+
+        # Does a stage already exist within the app pkg, or we need to create one?
+        # Using "if not exists" should take care of either case.
+        log.info("Checking if stage exists, or creating a new one if none exists.")
+        with self.use_role(role):
+            self._execute_query(
+                f"create schema if not exists {app_pkg}.{schema_and_stage.split('.')[0]}"
+            )
+            self._execute_query(
+                f"""
+                    create stage if not exists {app_pkg}.{schema_and_stage}
+                    encryption = (TYPE = 'SNOWFLAKE_SSE')"""
+            )
+
+        # Perform a diff operation and display results to the user for informational purposes
+        log.info(
+            f"Performing a diff between the Snowflake stage and your local deploy_root {self.deploy_root} directory."
+        )
+        diff: DiffResult = stage_diff(self.deploy_root, f"{app_pkg}.{schema_and_stage}")
+        log.info("Listing results of diff:")
+        print_result(
+            ObjectResult(
+                {
+                    "only on local": ", ".join(diff.only_local),
+                    "only on stage": ", ".join(diff.only_on_stage),
+                    "modified/unknown": ", ".join(diff.different),
+                    "identical": ", ".join(diff.identical),
+                }
+            )
+        )
+
+        # Upload diff-ed files to app pkg stage
+        log.info(
+            f"Uploading diff-ed files from your local {self.deploy_root} directory to the Snowflake stage."
+        )
+        sync_local_diff_with_stage(
+            role=role,
+            deploy_root_path=self.deploy_root,
+            diff_result=diff,
+            stage_path=f"{app_pkg}.{schema_and_stage}",
+        )
+
+    def app_run(self) -> SnowflakeCursor:
+
+        project_name = clean_identifier(self.definition["name"])
+        app_pkg = self.definition.get("package", {}).get(
+            "name", default_app_package(project_name)
+        )
+        app_pkg_role = self.definition.get("package", {}).get("role", default_role())
+
+        with self.use_role(app_pkg_role):
+            show_cursor = self._execute_query(
+                f"show application packages like '{app_pkg}'",
+                cursor_class=DictCursor,
+            )
+
+            if show_cursor.rowcount == 0:
+                # Create an app pkg, with distribution = internal to avoid triggering security scan
+                log.info(f"Creating new application package {app_pkg} in account.")
+                self._execute_query(
+                    f"""
+                    create application package {app_pkg}
+                        comment = {SPECIAL_COMMENT}
+                        distribution = internal
+                """
+                )
+            elif show_cursor.rowcount > 0:
```

Comment:
```suggestion
            else:
```
Or can we get negative values?